tasks/main.yml

```diff
@@ -0,0 +1,38 @@
+- name: set authorized keys
+  authorized_key:
+    user: "{{ passwordless_ssh_user }}"
+    state: present
+    key: "{{ lookup('file', passwordless_ssh_public_key) }}"
+  become: "{{ passwordless_ssh_become }}"
+  become_user: "{{ passwordless_ssh_become_user }}"
+
+- name: copy private key to remote SSH dir
+  copy:
+    src: "{{ passwordless_ssh_private_key }}"
+    dest: "{{ passwordless_ssh_user_home }}/.ssh/\
+           {{ passwordless_ssh_remote_key_file }}"
+    mode: 0600
+    owner: "{{ passwordless_ssh_user }}"
+  become: "{{ passwordless_ssh_become }}"
+  become_user: "{{ passwordless_ssh_become_user }}"
+
+- name: update known hosts with the other hosts in this current group
+  shell: ssh-keyscan {{ hostvars[item]['ansible_host'] |
+         default(hostvars[item]['inventory_hostname']) }} >>
+         {{ passwordless_ssh_user_home }}/.ssh/known_hosts
```

Comment:
Eventually, someone is going to have a` passwordless_ssh_user_home` with a space in the path, so this should probably get quoted.

Reply:
Is it possible to use the known_hosts module here?

https://docs.ansible.com/ansible/latest/modules/known_hosts_module.html

Reply:
Those people should be defenestrated from high places, but I will add quotes here.

Reply:
Totally agreed, but you know it's gonna happen.  :dagger: